pkg/proxy/integrations/httpparser/httpparser.go

```diff
@@ -489,11 +491,37 @@ func decodeOutgoingHttp(requestBuffer []byte, clientConn, destConn net.Conn, h *
 		if err != nil {
 			logger.Error("error while matching http mocks", zap.Any("metadata", getReqMeta(req)), zap.Error(err))
 		}
-
 		if !isMatched {
 			passthroughHost := false
-			for _, host := range models.PassThroughHosts {
-				if req.Host == host {
+			passThrough := models.PassThroughHosts
+			portPassThrough := []models.Filters{}
+			for _, filters := range h.GetPassThroughHosts().Filters {
+				if filters.Port == 0 {
+					passThrough = append(passThrough, filters.Host, filters.Path)
+				} else {
+					portPassThrough = append(portPassThrough, filters)
+				}
+			}
+			// Define the regular expression pattern
+
+			for _, host := range passThrough {
+				regex, err := regexp.Compile(host)
+				if err != nil {
+					continue
+				}
+				matches := regex.MatchString(req.URL.String())
+				if matches && host != "" || req.Host == host {
+					passthroughHost = true
```

Comment:
Shouldn't we have a break statement here..?

Reply:
done so